panic: interface conversion: interface {} is nil, not map[string]interface {}           batch.needsJobDefUpdate

### Terraform Core Version

1.8.0

### AWS Provider Version

5.60.0, 5.61.0

### Affected Resource(s)

* aws_batch_job_definition

### Expected Behavior

Plan should succeed

### Actual Behavior

```console
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 761 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/batch.needsJobDefUpdate(0xc003e70800)
```

### Relevant Error/Panic Output Snippet

```console
Stack trace from the terraform-provider-aws_v5.61.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 761 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/batch.needsJobDefUpdate(0xc003e70800)
github.com/hashicorp/terraform-provider-aws/internal/service/batch/job_definition.go:569 +0x1074
github.com/hashicorp/terraform-provider-aws/internal/service/batch.jobDefinitionCustomizeDiff({0x13f75fa0?, 0x208b4340?}, 0xc003e70800, {0xe?, 0xc002598340?})
github.com/hashicorp/terraform-provider-aws/internal/service/batch/job_definition.go:464 +0x3a
github.com/hashicorp/terraform-provider-aws/internal/service/batch.ResourceJobDefinition.Sequence.func23({0x170b2cc8, 0xc003cdf080}, 0xc003e70800, {0x14e571c0, 0xc002598340})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/customdiff/compose.go:69 +0x84
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).CustomizeDiff.func5({0x170b2cc8?, 0xc0023c5dd0?}, 0xc003e70800, {0x14e571c0, 0xc002598340})
github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:186 +0x63
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc000beb380, {0x170b2cc8, 0xc0023c5dd0}, 0xc001a2f1e0, 0xc0001b51d0, 0xc001078b88, {0x14e571c0, 0xc002598340}, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/schema.go:698 +0x4b4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x170b30f8?, {0x170b2cc8?, 0xc0023c5dd0?}, 0xc001a2f1e0, 0xc0023c5e00?, {0x14e571c0?, 0xc002598340?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:990 +0xdb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc003697068, {0x170b2cc8?, 0xc0023c5ce0?}, 0xc000deda90)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:858 +0xbe8
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).PlanResourceChange(0xc0014921c0, {0x170b2cc8?, 0xc0023c5a10?}, 0xc000deda90)
github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_PlanResourceChange.go:73 +0x2ad
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc001997cc0, {0x170b2cc8?, 0xc0023c51d0?}, 0xc0006d3d80)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:825 +0x3f0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0x14c3fb20, 0xc001997cc0}, {0x170b2cc8, 0xc0023c51d0}, 0xc0006d3d00, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:500 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a60200, {0x170b2cc8, 0xc0023c5140}, {0x17116620, 0xc000002000}, 0xc001f7c6c0, 0xc0025afd40, 0x208231e0, 0x0)
google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000a60200, {0x17116620, 0xc000002000}, 0xc001f7c6c0)
google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 41
google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.61.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

...terraform config...

### Steps to Reproduce

Our apply process occurs in a Terraform Enterprise workspace.

terraform apply

### Debug Output

_No response_

### Panic Output

```console
Stack trace from the terraform-provider-aws_v5.61.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 761 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/batch.needsJobDefUpdate(0xc003e70800)
github.com/hashicorp/terraform-provider-aws/internal/service/batch/job_definition.go:569 +0x1074
github.com/hashicorp/terraform-provider-aws/internal/service/batch.jobDefinitionCustomizeDiff({0x13f75fa0?, 0x208b4340?}, 0xc003e70800, {0xe?, 0xc002598340?})
github.com/hashicorp/terraform-provider-aws/internal/service/batch/job_definition.go:464 +0x3a
github.com/hashicorp/terraform-provider-aws/internal/service/batch.ResourceJobDefinition.Sequence.func23({0x170b2cc8, 0xc003cdf080}, 0xc003e70800, {0x14e571c0, 0xc002598340})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/customdiff/compose.go:69 +0x84
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).CustomizeDiff.func5({0x170b2cc8?, 0xc0023c5dd0?}, 0xc003e70800, {0x14e571c0, 0xc002598340})
github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:186 +0x63
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc000beb380, {0x170b2cc8, 0xc0023c5dd0}, 0xc001a2f1e0, 0xc0001b51d0, 0xc001078b88, {0x14e571c0, 0xc002598340}, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/schema.go:698 +0x4b4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x170b30f8?, {0x170b2cc8?, 0xc0023c5dd0?}, 0xc001a2f1e0, 0xc0023c5e00?, {0x14e571c0?, 0xc002598340?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:990 +0xdb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc003697068, {0x170b2cc8?, 0xc0023c5ce0?}, 0xc000deda90)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:858 +0xbe8
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).PlanResourceChange(0xc0014921c0, {0x170b2cc8?, 0xc0023c5a10?}, 0xc000deda90)
github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_PlanResourceChange.go:73 +0x2ad
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc001997cc0, {0x170b2cc8?, 0xc0023c51d0?}, 0xc0006d3d80)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:825 +0x3f0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0x14c3fb20, 0xc001997cc0}, {0x170b2cc8, 0xc0023c51d0}, 0xc0006d3d00, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:500 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a60200, {0x170b2cc8, 0xc0023c5140}, {0x17116620, 0xc000002000}, 0xc001f7c6c0, 0xc0025afd40, 0x208231e0, 0x0)
google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000a60200, {0x17116620, 0xc000002000}, 0xc001f7c6c0)
google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 41
google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.61.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Important Factoids

apply runs successfully if there is no jobdefinition. Fails when there are existing AWS Batch setup.

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/38716.txt
+++ b/.changelog/38716.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_batch_job_definition: Fix panic when checking `eks_properties` for job updates
+```

--- a/internal/service/batch/job_definition.go
+++ b/internal/service/batch/job_definition.go
@@ -516,7 +516,7 @@ func needsJobDefUpdate(d *schema.ResourceDiff) bool {
 		}
 
 		var oeks, neks *batch.EksPodProperties
-		if len(o.([]interface{})) > 0 && 0.([]interface{})[0] != nil {
+		if len(o.([]interface{})) > 0 && o.([]interface{})[0] != nil {
 			oProps := o.([]interface{})[0].(map[string]interface{})
 			if opodProps, ok := oProps["pod_properties"].([]interface{}); ok && len(opodProps) > 0 {
 				oeks = expandEKSPodProperties(opodProps[0].(map[string]interface{}))

--- a/internal/service/batch/job_definition.go
+++ b/internal/service/batch/job_definition.go
@@ -516,14 +516,14 @@ func needsJobDefUpdate(d *schema.ResourceDiff) bool {
 		}
 
 		var oeks, neks *batch.EksPodProperties
-		if len(o.([]interface{})) > 0 {
+		if len(o.([]interface{})) > 0 && 0.([]interface{})[0] != nil {
 			oProps := o.([]interface{})[0].(map[string]interface{})
 			if opodProps, ok := oProps["pod_properties"].([]interface{}); ok && len(opodProps) > 0 {
 				oeks = expandEKSPodProperties(opodProps[0].(map[string]interface{}))
 			}
 		}
 
-		if len(n.([]interface{})) > 0 {
+		if len(n.([]interface{})) > 0 && n.([]interface{})[0] != nil {
 			nProps := n.([]interface{})[0].(map[string]interface{})
 			if npodProps, ok := nProps["pod_properties"].([]interface{}); ok && len(npodProps) > 0 {
 				neks = expandEKSPodProperties(npodProps[0].(map[string]interface{}))
@@ -540,12 +540,12 @@ func needsJobDefUpdate(d *schema.ResourceDiff) bool {
 		}
 
 		var ors, nrs *batch.RetryStrategy
-		if len(o.([]interface{})) > 0 {
+		if len(o.([]interface{})) > 0 && o.([]interface{})[0] != nil {
 			oProps := o.([]interface{})[0].(map[string]interface{})
 			ors = expandRetryStrategy(oProps)
 		}
 
-		if len(n.([]interface{})) > 0 {
+		if len(n.([]interface{})) > 0 && n.([]interface{})[0] != nil {
 			nProps := n.([]interface{})[0].(map[string]interface{})
 			nrs = expandRetryStrategy(nProps)
 		}
@@ -560,12 +560,12 @@ func needsJobDefUpdate(d *schema.ResourceDiff) bool {
 		}
 
 		var ors, nrs *batch.JobTimeout
-		if len(o.([]interface{})) > 0 {
+		if len(o.([]interface{})) > 0 && o.([]interface{})[0] != nil {
 			oProps := o.([]interface{})[0].(map[string]interface{})
 			ors = expandJobTimeout(oProps)
 		}
 
-		if len(n.([]interface{})) > 0 {
+		if len(n.([]interface{})) > 0 && n.([]interface{})[0] != nil {
 			nProps := n.([]interface{})[0].(map[string]interface{})
 			nrs = expandJobTimeout(nProps)
 		}
@@ -616,7 +616,7 @@ func resourceJobDefinitionCreate(ctx context.Context, d *schema.ResourceData, me
 			}
 		}
 
-		if v, ok := d.GetOk("eks_properties"); ok && len(v.([]interface{})) > 0 {
+		if v, ok := d.GetOk("eks_properties"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 			eksProps := v.([]interface{})[0].(map[string]interface{})
 			if podProps, ok := eksProps["pod_properties"].([]interface{}); ok && len(podProps) > 0 {
 				if aws.StringValue(input.Type) == batch.JobDefinitionTypeContainer {

